src/main/java/com/jagrosh/vortex/commands/moderation/ModlogsCmd.java

```diff
@@ -21,6 +21,64 @@ public ModlogsCmd(Vortex vortex) {
         this.guildOnly = true;
     }
 
+    @Override
+    protected void execute(CommandEvent event) {
+        if (event.getArgs().equalsIgnoreCase("help")) {
+            event.replySuccess("This command is used to see a user's modlogs for the current server. Please mention a user or use a user ID to check their modlogs.");
+            return;
+        }
+
+        long id = CommandTools.getPossibleUserId(event.getArgs().trim());
+
+        if (id == -1) {
+            if (event.getArgs().trim().isEmpty()) {
+                id = event.getAuthor().getIdLong();
+            } else {
+                event.reply("Please mention someone or enter a valid user ID");
+                return;
+            }
+        }
+
+        List<Modlog> modlogs = Database.getAllModlogs(event.getGuild().getIdLong(), id);
+
+        int size = modlogs.size();
+
+        if (size == 0) {
+            event.reply("Could not find any modlogs for that user");
+            return;
+        }
+
+        EmbedBuilder[] embeds = new EmbedBuilder[size / 25 + 1];
+        for (int i = 0; i < embeds.length; i++) {
+            embeds[i] = new EmbedBuilder();
+        }
+
+        if (event.getJDA().getUserById(id) != null) {
+            embeds[0].setAuthor(String.format("%d modlog%s found for %s#%s (%d)", size, size == 1 ? "" : "s", event.getJDA().getUserById(id).getName(), event.getJDA().getUserById(id).getDiscriminator(), id), null, String.format(event.getJDA().getUserById(id).getAvatarUrl()));
+        } else {
+            embeds[0].setAuthor(String.format("%d modlog%s found for %d", size, size == 1 ? "" : "s", id), null, String.format(event.getJDA().getUserById(id).getAvatarUrl()));
```

Comment:
This block of code only executes when `event.getJda().getUserById(id)` is null. By default this library gets users/guilds/channels etc from the bots cache, which can lead to null pointer exceptions. Most `getXxx(id)` methods have a `retrieveXxx(id)` counterpart to make api requests ([helpful article](https://jda.wiki/using-jda/using-restaction/))

Reply:
Although it may return null if the user is deleted I think? Not sure what to do for pfps in that case

Reply:
There's no issues changing it to setAuthor though right? I used setAuthor to let the avatar icon be in line with the title section

Reply:
Yeah I think setAuthor is only visual